types/serverless-step-functions/types/task.d.ts

```diff
@@ -0,0 +1,25 @@
+import { Catch, Retry } from './errors';
+import { IntrinsicFunction, JsonObject, JsonPath, PositiveInteger, Resource } from './state';
+
+// https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-task-state.html#task-state-fields
+export interface Task {
+    Type: 'Task';
+    Comment?: string;
+    InputPath?: JsonPath;
+    OutputPath?: JsonPath;
+    Next?: string;
+    End?: boolean;
+    Resource: Resource;
+    Parameters?: JsonObject;
+    Credentials?: string | IntrinsicFunction | JsonPath;
+    ResultPath?: JsonPath;
+    ResultSelector?: JsonObject;
+    Retry?: Retry[];
+    Catch?: Catch[];
+    TimeoutSeconds?: PositiveInteger;
+    TimeoutSecondsPath?: JsonPath;
+    HeartbeatSeconds?: PositiveInteger;
+    HeartbeatSecondsPath?: JsonPath;
```

Comment:
This looks good but it allows for invalid combinations of properties that could be enforced by the type definition.

For example:
- a `Task` must have a `Next` or `End` but not both
- a `Task` may have a `TimeoutSeconds` or `TimeoutSecondsPath` but not both
- a `Task` may have a `HeartbeatSeconds` or `HeartbeatSecondsPath` but not both

These rules could be encoded into the type like so: 

```typescript
export type Task = {
    Type: 'Task';
    Comment?: string;
    InputPath?: JsonPath;
    OutputPath?: JsonPath;
    Resource: Resource;
    Parameters?: JsonObject;
    Credentials?: string | IntrinsicFunction | JsonPath;
    ResultPath?: JsonPath;
    ResultSelector?: JsonObject;
    Retry?: Retry[];
    Catch?: Catch[];
} & EndOrNext & TimeoutSecondsOrPath & HeartbeatSecondsOrPath


export type EndOrNext = {
    Next: string;
    End?: never;
} | {
    Next?: never;
    End: boolean;
}

export type TimeoutSecondsOrPath = {
    TimeoutSeconds?: PositiveInteger;
    TimeoutSecondsPath?: never;
} | {
    TimeoutSeconds?: never;
    TimeoutSecondsPath?: JsonPath;
}

export type HeartbeatSecondsOrPath = {
    HeartbeatSeconds?: PositiveInteger;
    HeartbeatSecondsPath?: never;
} | {
    HeartbeatSeconds?: never;
    HeartbeatSecondsPath?: JsonPath;
}

```

Reply:
Those change look good although I couldn't test them. Also I didn't know that could be done.